### PR TITLE
Customizable dialog content

### PR DESCRIPTION
**Pull request by albinpoignot**
**tl;dr;**
- Allow user to set custom string
- Remove hard-coded string and put them in `strings.xml`

**Long version**

The current version of Gandalf uses hard-coded string for dialog's title and button. This does not allow customization neither translation. 

This PR offers the possibility to set custom Strings when "installing" Gandalf. It also removes hard-coded string to put them in the `strings.xml` file.

The class `DialogStringsHolder` stores the custom strings used for AlertDialog in the `BootstrapDialogUtil` class.

### CODE REVIEW

**Comment by btkelly** [comment]
I like this, good to make it as customizable as possible. As far as translations go, do you think it would be helpful to provide a set of common languages as part of the library or should the consumer just use this new DialogStringsHolder for translations?

File: gandalf/src/main/java/io/github/btkelly/gandalf/holders/DialogStringsHolder.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    private String skipUpdateButton;
+    private String closeAppButton;
+    private String okButton;
+
+    public DialogStringsHolder(Context context) {
+        this.context = context;
+    }
+
+    public String getUpdateAvailableTitle() {
+        if (updateAvailableTitle == null) {
```
Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/holders/DialogStringsHolder.java]: Might be good to use the StringUtils.isBlank method here as it checks for an empty string as well.

Comment (albinpoignot) [inline gandalf/src/main/java/io/github/btkelly/gandalf/holders/DialogStringsHolder.java]: Ok, will change it :)

**Comment by albinpoignot** [comment]
I tried using a new string.xml to add my own language. However, I didn't succeed the first time (I don't know why !) and I do not have more time to investigate right now.

File: gandalf/src/main/java/io/github/btkelly/gandalf/holders/DialogStringsHolder.java (hunk 1 above)
Comment (albinpoignot) [inline gandalf/src/main/java/io/github/btkelly/gandalf/holders/DialogStringsHolder.java]: I just checked the `isBlank()` code. If I use it, the user will not be able to set an empty title (empty String). Even if this seems to be silly, I think we should let the user make whatever he wants. Do you think we could stay with a "simple null-check" here ?

Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/holders/DialogStringsHolder.java]: Ah, yeah probably a good call, we can leave the null check

**Comment by btkelly** [comment]
If you can merge this with master we should be good to go

**Comment by btkelly** [comment]
Approved

**Pull request merged**